bin/oneoff/backfill_data/user_cap_state_columns

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env ruby
+
+# This script migrates CAP data from properties to separate indexed columns.
+
+require 'ruby-progressbar'
+require_relative '../../../dashboard/config/environment'
+
+max_id = User.maximum(:id)
+batch_size = 100_000
+total_updated = 0
+
+progress_bar_title = proc {|num_updates = 0| "Processed[%c/%C]: |%W| Updated: #{total_updated += num_updates} %a"}
```

Comment:
In addition to a live progress bar, we should adopt the best practices for logging successful and failed batches from this data clean up script: https://github.com/code-dot-org/code-dot-org/blob/staging/bin/oneoff/data_fix/fix_invalid_users

Reply:
The script is straightforward and doesn't require logging failed batches.
If the script is interrupted, it can always be restarted from where it stopped due to this condition: `where(cap_state: nil)`

Reply:
cc @Hamms

Reply:
We have inadvertently corrupted production data in the past with data update scripts, so careful logging of the changes that a back fill script has made including detailed error logging of each failed batch is important. The `fix_invalid_users` script has some of the best logging / tracking logic that we've implemented in recent years.

Reply:
What data corruption are you referring to? Because the worst that can happen with this script is that some records might not be updated.

Reply:
What data corruption are you referring to? The worst that can happen with this script is that some records might not be updated. However, this is not an issue because the script will exclude the already updated records when rerun

Reply:
Hopefully the script and its logic don't have any defects, but when we're modifying production data in a critical table and particularly when we don't have unit tests for the logic that finds and updates records, we need to be very careful. Logging each batch and each record (`users.id`) that we've successfully updated and that we've failed to update is important.